project-base/storefront/hooks/comparison/useComparison.tsx

```diff
@@ -18,28 +18,23 @@ export const useComparison = () => {
     const [, removeProductFromComparison] = useRemoveProductFromComparisonMutationApi();
     const [, cleanComparison] = useCleanComparisonMutationApi();
     const comparisonUuid = usePersistStore((store) => store.comparisonUuid);
-    const updateUserState = usePersistStore((store) => store.updateUserState);
+    const updateComparisonUuid = usePersistStore((store) => store.updateComparisonUuid);
     const [isPopupCompareOpen, setIsPopupCompareOpen] = useState(false);
-    const initialLoadRef = useRef(false);
-    const [isLoadingVisible, setIsLoadingVisible] = useState(true);
+    const [isFetchingPaused, setIsFetchingPaused] = useState(true);
 
     const [{ data: comparisonData, fetching }] = useComparisonQueryApi({
         variables: { comparisonUuid },
-        pause: !comparisonUuid && !isUserLoggedIn,
+        pause: isFetchingPaused,
     });
 
     useEffect(() => {
-        if (initialLoadRef.current) {
-            initialLoadRef.current = true;
-        } else {
-            setIsLoadingVisible(fetching);
-        }
-    }, [fetching]);
+        setIsFetchingPaused(!comparisonUuid && !isUserLoggedIn);
+    }, [comparisonUuid]);
```

Comment:
Because there was a question regarding this on Slack, I am just asking to make sure I understand your motivation. This is done so that the fetching paused is updated AFTER the first render through the useEffect and the initial value is true, right? Can you elaborate why that is necessary?

Reply:
Yup, motivation is exactly this. Since comparisonUuid is in persist store in browser this information is not accessible to Next server. So it would cause Next complaining about different rendered UI.